tutor/src/components/buttons/save-practice.js

```diff
@@ -94,30 +95,30 @@ const SavePracticeButton = observer(({
     return Boolean(taskStep && taskStep.multiPartGroup);
   };
 
-  const savePracticeButton = (
-    <>
-      {
-        disabled
-          ? (
-            <StyledSavePracticeButton
-              disabled={disabled}
-              className="save-practice-button"
-              data-test-id="save-practice-button"> 
-              {getIconAndLabel()}
-            </StyledSavePracticeButton>
-          )
-          : (
-            <StyledSavePracticeButton
-              onClick={saveOrRemovePracticeQuestion}
-              isSaved={isSaved()}
-              className="save-practice-button"
-              data-test-id="save-practice-button"> 
-              {getIconAndLabel(isSaved(), practiceQuestions.isAnyPending)}
-            </StyledSavePracticeButton>
-          )
-      }
-    </>
-  );
+  const savePracticeButton = () => {
+    if(disabled) {
+      return (
+        <StyledSavePracticeButton
+          disabled={disabled}
+          className="save-practice-button"
+          data-test-id="save-practice-button"> 
+          {getIconAndLabel()}
+        </StyledSavePracticeButton>
+      );
+    }
+
+    return (
+      <StyledSavePracticeButton
+        onClick={saveOrRemovePracticeQuestion}
+        isSaved={isSaved()}
+        className="save-practice-button"
+        data-test-id="save-practice-button"> 
+        {getIconAndLabel(isSaved(), practiceQuestions.isAnyPending)}
+      </StyledSavePracticeButton>
+    );
+  };
+    
```

Comment:
nice cleanup. this reads much easier to me.

Reply:
thanks. I changed it mainly because the `<></>` blocks the render of the tooltip.